### ISSUE TITLE
VirtualScroll with lazy load, Calling reset does nothing.

### ISSUE THREAD

**Issue by olingerc** [primefaces/primeng#2955]
```
[x ] bug report => Search github for a similar issue or PR before submitting
[ ] feature request => Please check if request is not on the roadmap already https://github.com/primefaces/primeng/wiki/Roadmap
[ ] support request => Please do not submit support request here, instead see http://forum.primefaces.org/viewforum.php?f=35
```

In case one uses virtualScroll plus lazy loading, calling reset() does not do anything. [This line](https://github.com/primefaces/primeng/blob/master/components/datatable/datatable.ts#L2019) should also check for virtualScroll. i.e.
```javascript
if(this.paginator || this.virtualScroll) {
    this.paginate({
        first: 0,
        rows: this.rows
    });
}
```

* **PrimeNG version:** 4.0.2

**Comment by izll** [comment]
+1

**Comment by cagataycivici** [comment]
Fixed via #3128
